entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/SiteEntityProvider.java

```diff
@@ -903,10 +907,24 @@ public void updateEntity(EntityReference ref, Object entity, Map<String, Object>
             s.setSkin(site.getSkin());
             s.setTitle(site.getTitle());
 
-            // put in properties if admin, otherwise ignore
+            // put in properties if admin, otherwise allow update of specific configurable fields.
             if (admin) {
                 ResourcePropertiesEdit rpe = s.getPropertiesEdit();
                 rpe.set(site.getProperties());
+            } else {
+                if (updateableSiteProps != null && updateableSiteProps.length != 0) {
+                    ResourcePropertiesEdit rpe = s.getPropertiesEdit();
+                    Iterator it = site.getProperties().getPropertyNames();
+                    for (String prop : updateableSiteProps) {
+                        while (it.hasNext()) {
```

Comment:
I think you want to reset the iterator inside here don't you? Otherwise you're dependent on the order of the configuration and the iterator order. But thinking a little is there a reason why you don't just call `site.getProperties().getProperty(prop)` and check to see if it's `null`?